subt/main.py

```diff
@@ -896,7 +904,9 @@ def play_virtual_track(self):
         self.stdout('Using times', times_sec)
 
         # add extra sleep to give a chance to the other robot (based on name)
-        self.wait(timedelta(seconds=times_sec[0]), use_sim_time=True)
+        # and making sure that the sleep time is non-trivial so that ROS modules
+        # can start in the meantime.
+        self.wait(timedelta(seconds=max(5, times_sec[0])), use_sim_time=True)
```

Comment:
I am not very exited by this. Can we instead wait on some output of the said ROS modules? Like we do above where we say we "wait for critical data". BTW: it is also the reason why the test fails.

Reply:
`pose3d` is such critical data, but we do not produce it in all robots, so waiting for it might be tricky.

Reply:
I think drone was the only one, wasn't it?

Reply:
Yes.

Thinking about this more ... If we start with `self.xyz = None` instead of `self.xyz = (0, 0, 0)` in `__init__()`, we could wait for `self.xyz` being set. That would happen either in `on_pose2d` or in `on_pose3d`. That _could_ work across all the robots. It would signal that whatever our source of odometry is (ROS, or some microcontroller), it is ready.

Reply:
I am trying to fix it now - note that also the simulation and change in `on_pose2d` when offset is not ready needs to be done - WIP (next 20min, I hope)

Reply:
The final solution is waiting on `self.xyz`.